lib/IDE/ModuleInterfacePrinting.cpp

```diff
@@ -686,14 +696,15 @@ void swift::ide::printModuleInterface(
 
     for (auto CM : ClangModules) {
       for (auto DeclAndLoc : ClangDecls[CM.first])
-        PrintDecl(DeclAndLoc.first);
+        if (PrintDecl(DeclAndLoc.first) && Options.EmptyLineBetweenDecls)
+          Printer.printNewline();
     }
   }
 
   if (!(TraversalOptions & ModuleTraversal::SkipOverlay) || !TargetClangMod) {
     for (auto *D : SwiftDecls) {
-      if (PrintDecl(D))
-        Printer << "\n";
+      if (PrintDecl(D) && Options.EmptyLineBetweenDecls)
+        Printer.printNewline();
```

Comment:
This makes `swift-ide-test -print-interface` NOT print newlines between top-level decls.
Currently `EmptyLineBetweenDecls` (formerly `EmptyLineBetweenMembers`) is enabled in generated-interface only.